FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -21,27 +32,75 @@ import com.flagsmith.internal.enqueueWithResult
 class Flagsmith constructor(
     private val environmentKey: String,
     private val baseUrl: String = "https://edge.api.flagsmith.com/api/v1/",
+    private val eventSourceUrl: String? = null,
     private val context: Context? = null,
     private val enableAnalytics: Boolean = DEFAULT_ENABLE_ANALYTICS,
+    private val enableRealtimeUpdates: Boolean = false,
     private val analyticsFlushPeriod: Int = DEFAULT_ANALYTICS_FLUSH_PERIOD_SECONDS,
     private val cacheConfig: FlagsmithCacheConfig = FlagsmithCacheConfig(),
     private val defaultFlags: List<Flag> = emptyList(),
     private val requestTimeoutSeconds: Long = 4L,
     private val readTimeoutSeconds: Long = 6L,
-    private val writeTimeoutSeconds: Long = 6L
-) {
-    private val retrofit: FlagsmithRetrofitService = FlagsmithRetrofitService.create(
-        baseUrl = baseUrl, environmentKey = environmentKey, context = context, cacheConfig = cacheConfig,
-        requestTimeoutSeconds = requestTimeoutSeconds, readTimeoutSeconds = readTimeoutSeconds, writeTimeoutSeconds = writeTimeoutSeconds)
+    private val writeTimeoutSeconds: Long = 6L,
+    override var lastSeenAt: Double = 0.0 // from FlagsmithEventTimeTracker
+) : FlagsmithEventTimeTracker {
+    private lateinit var retrofit: FlagsmithRetrofitService
+    private var cache: Cache? = null
+    private var lastUsedIdentity: String? = null
     private val analytics: FlagsmithAnalytics? =
         if (!enableAnalytics) null
         else if (context != null) FlagsmithAnalytics(context, retrofit, analyticsFlushPeriod)
         else throw IllegalArgumentException("Flagsmith requires a context to use the analytics feature")
 
+    private val eventService: FlagsmithEventService? =
+        if (!enableRealtimeUpdates) null
+        else FlagsmithEventService(eventSourceUrl = eventSourceUrl, environmentKey = environmentKey) { event ->
+            if (event.isSuccess) {
+                lastEventUpdate = event.getOrNull()?.updatedAt ?: lastEventUpdate
+
+                // Check whether this event is anything new
+                if (lastEventUpdate > lastSeenAt) {
+                    // First evict the cache otherwise we'll be stuck with the old values
+                    cache?.evictAll()
```

Comment:
Because it's an HTTP cache a cache miss would just result in a GET for the flags. There will always be the possibility of two simultaneous GET requests that haven't yet seen a cache but the behaviour is well-defined